core/src/main/scala/neotypes/DeferredQuery.scala

```diff
@@ -4,7 +4,7 @@ import neotypes.mappers.{ExecutionMapper, ResultMapper}
 
 import scala.collection.mutable.StringBuilder
 
-private[neotypes] final case class DeferredQuery[T](query: String, params: Map[String, Any] = Map.empty) {
+final case class DeferredQuery[T] private[neotypes] (query: String, params: Map[String, Any] = Map.empty) {
```

Comment:
I forgot that making the class itself private, forbids creation of methods that return such instances, which is useful.
Thus, I make the class public again but made its constructor private.

The same goes for the Builder.